qiskit/providers/backend.py

```diff
@@ -229,3 +233,233 @@ class can handle either situation.
             Job: The job object for the run
         """
         pass
+
+
+class BackendV2(Backend, ABC):
+    """Abstract class for Backends
+
+    This abstract class is to be used for all Backend objects created by a
+    provider. This version differs from earlier abstract Backend classes in
+    that the configuration attribute no longer exists instead attributes
+
+    The ``options`` attribute of the backend is used to contain the dynamic
+    user configurable options of the backend. It should be used more for
+    runtime options that configure how the backend is used. For example,
+    something like a ``shots`` field for a backend that runs experiments which
+    would contain an int for how many shots to execute.
+
+    If migrating a provider from :class:`~qiskit.providers.BackendV1` or
+    :class:`~qiskit.providers.BaseBackend` one thing to keep in mind is for
+    backwards compatibility you might need to add a configuration method that
+    will build a :class:`~qiskit.providers.models.BackendConfiguration` object
+    and :class:`~qiskit.providers.models.BackendProperties` from the attributes
+    defined in this class for backwards compatibility.
+    """
+
+    version = 2
+
+    def __init__(self, provider, **fields):
+        self._options = self._default_options()
+        self._provider = provider
+        self._coupling_map = None
+        if fields:
+            for field in fields:
+                if field not in self._options.data:
+                    raise AttributeError("Options field %s is not valid for this backend" % field)
+            self._options.update_config(**fields)
+
+    @property
+    def instructions(self) -> List[Gate]:
+        """A list of Gate instances that the backend supports."""
+        return list(self.target.gates)
+
+    @property
+    def instructions_names(self) -> List[str]:
+        """A list of Gate names that the backend supports."""
+        return list(self.target.gate_names)
+
+    @property
+    @abstractmethod
+    def target(self):
```

Comment:
It is not so clear to me how a `backend` is different from a `target`?  Is this just to push the loading of the "properties" outside of the init of the `backend` like `backend.properties()` does today?

Reply:
The target is just a container data structure for representing the device constraints to the transpiler. It's basically just a subset of the backend's characteristics that the transpiler will use directly and are required to be able to use it as a target device for the compilation. It's basically to replace a bunch of separate things (like the basis gates list, the coupling map, instruction durations, etc) into a single data structure. At first I was thinking of just doing the nested dict inside a `Target` directly on the `BackendV2` but I figured having a separate class for it would make creating it easier then we could provide some caching, a fixed api for creating it (to ensure things are consistently formatted), etc. It's also a smaller target for serialization if that's a concern in the future.

As for how the backend creates it that's up to each provider implementation depending on how the backend api works. I expect the qiskit-ibm provider will need to adjust how it's doing the loading because target combines some elements from `BackendConfiguration` and `BackendProperties`, but target can be evaluated lazily after `__init__` if that's what makes sense.

Reply:
Yeah I am just thinking that already the properties can be ~40mb in size, and that will just keep growing.  thus if it is required at init then there is a lag.

Reply:
Those are just alias shortcuts for things people would get from the backend configuration previously (basis gates in that case). I was trying to avoid the two level access that existed before for the commonly accessed things.

At a higher level than my previous response the split is here for 2 reasons, the backend is a superset around what the target contains and offers. It includes methods for job submission and interacting with previous runs or querying an api for getting additional details (or historical details) as well as anything custom for a particular provider's backends that's not broadly used yet. While the target is just concerned about begin a data container representing a backend to the transpiler, it contains the information the transpiler requires or optionally can use when calling via `transpile()`. They need to be separate objects because of what I said before and how the transpiler works under the covers and that it may be running in multiple processes. `BackendV2` based objects are not guaranteed to pickle-able as they often have objects that contain compiled objects (in the case of simulators) or async handles or other references to API communication which can't be pickled (which are those extra responsibilities and why backends contain a target). 

Another way to think about it is the target is a terra native class that is essentially just the combination of the data previously contained in `configuration()`, `properties()`, and `defaults()` that can/is be used by `transpile()` and replaces `CouplingMap`, `InstructionDurations`, `InstructionScheduleMap`, `basis_gates`, etc with a single structure. While the `BackendV2` defines the abstract interface for how users interact with any backend and it contains a target so we can pass the data about the backend to the transpiler.